src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -466,19 +484,19 @@ class NBDWatchCtx : public librbd::UpdateWatchCtx
   }
 };
 
-static int open_device(const char* path, bool try_load_module = false)
+static int open_device(const char* path, Config *cfg = NULL, bool try_load_module = false)
```

Comment:
Ok, it can't crash in do_unmap case. But it in general, if you allow a pointer to be null (and setting this as default suggests it is a valid use case), you need to check it every time you do pointer dereference.

`open_device(path, nullptr, true)` will crash, and although this combination is unlikely to be ever used this looks not very good. Probably I would try to rewrite `open_device` to check cfg at early stage and return if it is null.

But leaving it as it is now is ok to me too. I suggest only to use `nullptr` instead of `NULL` in a new code.